### ISSUE TITLE
$error isn't set For Deeply Nested Collections

### ISSUE THREAD

**Issue by alexhernandez** [monterail/vuelidate#101]
On submission of empty form fields shouldn't the first $each[0] collection item have `$error = true` if `$invalid` is set to `true` ?

```javascript
{
  "form": {
    "service": {
      "required": false,
      "$invalid": true,
      "$dirty": false,
      "$error": false,
      "$pending": false,
      "$params": {
        "required": {
          "type": "required"
        }
      }
    },
    "rateUnit": {
      "required": false,
      "$invalid": true,
      "$dirty": false,
      "$error": false,
      "$pending": false,
      "$params": {
        "required": {
          "type": "required"
        }
      }
    },
    "rate": {
      "effectiveDate": {
        "required": false,
        "isDate": false,
        "$invalid": true,
        "$dirty": false,
        "$error": false,
        "$pending": false,
        "$params": {
          "required": {
            "type": "required"
          },
          "isDate": null
        }
      },
      "paymentMethodRates": {
        "required": true,
        "minLength": true,
        "$each": {
          "0": {
            "amount": {
              "required": false,
              "$invalid": true,
              "$dirty": false,
              "$error": false,
              "$pending": false,
              "$params": {
                "required": {
                  "type": "required"
                }
              }
            },
            "paymentType": {
              "required": false,
              "$invalid": true,
              "$dirty": false,
              "$error": false,
              "$pending": false,
              "$params": {
                "required": {
                  "type": "required"
                }
              }
            },
            "$invalid": true,
            "$dirty": false,
            "$error": false,
            "$pending": false,
            "$params": {
              "amount": null,
              "paymentType": null
            }
          },
          "$invalid": true,
          "$dirty": false,
          "$error": false,
          "$pending": false,
          "$params": {
            "0": null
          }
        },
        "$invalid": true,
        "$dirty": false,
        "$error": false,
        "$pending": false,
        "$params": {
          "required": {
            "type": "required"
          },
          "minLength": {
            "type": "minLength",
            "min": 1
          },
          "$each": null
        }
      },
      "$invalid": true,
      "$dirty": false,
      "$error": false,
      "$pending": false,
      "$params": {
        "effectiveDate": null,
        "paymentMethodRates": null
      }
    },
    "$invalid": true,
    "$dirty": false,
    "$error": false,
    "$pending": false,
    "$params": {
      "service": null,
      "rateUnit": null,
      "rate": null
    }
  },
  "$invalid": true,
  "$dirty": false,
  "$error": false,
  "$pending": false,
  "$params": {
    "form": null
  }
}
```

**Comment by alexhernandez** [comment]
Seems like the issue also occurs on the Vuelidate docs example https://monterail.github.io/vuelidate/#sub-collections-validation

The issue seems to be that firing `this.$v.form.$touch()` on the parent doesn't impact `$each` children

**Comment by Frizi** [comment]
This is a duplicate of #95. Addressed in #96.

**Comment by alexhernandez** [comment]
@Frizi You Awesome Son of a Gun. Thank You sir.
